### DISCUSSION TITLE
How i can destroy created zustand store

### DISCUSSION THREAD

**Discussion by artshakhov** [pmndrs/zustand#2283] [Q&A]
Hi everyone, i need help to destroy already created zustand store.
I use zustand as replacement of useState, for implementation local state of some feature (some large module). I create context and use it with new zustand store for each instance of some feature (for example calculator). All is work fine, but i cant destroy store when some instance is unmount, StoreApi.destroy mark as deprecated. If i call him nothing happens (I also see created instance in redux devtools)

**Comment by dai-shi** [comment]
Zustand store with no subscriptions is garbage collectable.
The only issue is with devtools. See also: #2254 

If your concern is production, the easiest solution is `useFooStore.setState(null)`.

**Reply by artshakhov** [reply]
Can i use some method to see list of actual stores, and make sure that havent bugs with subscriptions?
setState null good idea, i try it)

**Reply by dai-shi** [reply]
unfortunately, we don't have a native way to count subscriptions. You need some workaround like this:
```js
const useFooStore = create(...);
const origSubscribe = useFooStore.subscribe;
let subscriptions = 0;
useFooStore.subscribe = (listener) => {
  ++subscriptions;
  const unsub = origSubscribe(listener);
  return () => { unsub(); --subscriptions; };
};
```
